lib/Controller/RemoteController.php

```diff
@@ -534,4 +534,76 @@ public function exceptionResponse(
 			($e->getCode() > 0) ? $e->getCode() : $httpErrorCode
 		);
 	}
+
+
+	/**
+	 * use this one if a method from a Controller is only PublicPage when remote client asking for Json
+	 *
+	 * try {
+	 *      $this->publicPageJsonLimited();
+	 *      return new DataResponse(['test' => 42]);
+	 * } catch (JsonNotRequestedException $e) {}
+	 *
+	 *
+	 * @throws NotLoggedInException
+	 * @throws JsonNotRequestedException
+	 */
+	private function publicPageJsonLimited(): void {
+		if (!$this->jsonRequested()) {
+			if (!OC::$server->get(IUserSession::class)
+							->isLoggedIn()) {
+				throw new NotLoggedInException();
+			}
+
+			throw new JsonNotRequestedException();
+		}
+	}
+
+
+	/**
+	 * @return bool
+	 */
+	private function jsonRequested(): bool {
+		return ($this->areWithinAcceptHeader(
+			[
+				'application/json',
+				'application/ld+json',
+				'application/activity+json'
+			]
+		));
+	}
+
+
+	/**
+	 * @param array $needles
+	 *
+	 * @return bool
+	 */
+	private function areWithinAcceptHeader(array $needles): bool {
+		$request = OC::$server->get(IRequest::class);
+		$accepts = array_map([$this, 'trimHeader'], explode(',', $request->getHeader('Accept')));
+
+		foreach ($accepts as $accept) {
+			if (in_array($accept, $needles)) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	/**
```

Comment:
php doc comments that just repeat the type information of the method are useless